### PR TITLE
Add Complex Filter for Canvas

### PR DESCRIPTION
**Pull request by kaioken**
- [x] - Update search filter
- [x] - Add custom fields
- [x] - Add complex filters for httpd

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     * @return void
+     */
+    public function findBySql(string $sql): array
+    {
+        $client = new GuzzleClient([
+            'base_uri' => $this->host,
+        ]);
+
+        // since 6.x we need to use POST
+        $response = $client->post('/_sql', [
```
Comment (gabbanaesteban) [inline src/Client.php:36]: @kaioken Put this inside of try-catch, if the URL can't be reached or the connection can't be established, an exception will be thrown.

Comment (kaioken) [reply src/Client.php:36]: By default guzzle throws the exception I was planning on capturing it on the API controller

Comment (gabbanaesteban) [reply src/Client.php:36]: Yes, `an exception will be thrown`.
I think it should be handled in the package itself and throw a custom elasticsearch-package-exception. In that way, whoever uses the package will get an Elastic related exception and will determinate how to handle it. 
For example, if the S3 SDK throw an exception and I'm using flysystem, the exception that I get in my app is from flysystem not from S3 SDK.
I think you can have more control and context if the exception is from the package you are using and not the packages the library use.

Comment (kaioken) [reply src/Client.php:36]: Will see on how to handle it than wen I get to the complex canvas filter

**Pull request merged**